scripts/organismSpecies/parse_species.py

```diff
@@ -0,0 +1,203 @@
+           CODE_TO_KINGDOM[info['kingdom']] + '\n' + 'uniProtOrganismCode: "' +
+           code + '"\n' + 'dcid: "bio/' + dcid + '"\n')
+
+    return mcf, name_seen
+
+
+def get_class_name(a_string):
+    """Convert a name string to format: ThisIsAnUnusualName.
+    Take a space delimited string, return a class name such as ThisIsAnUnusualName
+    Here we use this function for instance name. Thus it allows to start with a number
+    """
+    joint_name = a_string.title().replace(' ', '')
+    # substitute except for  _, character, number
+    non_legitimate = re.compile(r'[\W]+')
+    class_name = non_legitimate.sub('', joint_name)
+    return class_name
+
+
+def get_mcf(combine_lines, seen_name_to_dcid):
+    """Generate mcf list.
+    Args:
+        combine_lines: a list, each contains a line from the database
+        seen_name_to_dcid: a dict mapping seen species name to dcid
+    Returns:
+        mcf_seen_list: new schema list of the species already in DC
+        mcf_list: new schema list of the species not in DC
+    """
+
+    mcf_list = []
+    mcf_seen_list = []
+    code = None
+    name_type_map = {'N': 'Official', 'C': 'Common', 'S': 'Alternate'}
+    CODE_TO_KINGDOM = {
+        'A': 'Archaea',
+        'B': 'Bacteria',
+        'E': 'Eukaryota',
+        'V': 'Virus',
+        'O': 'OtherOrganismKingdom'
+    }
+    info = {}
+    # Original data format:
+    # AADNV V  648330: N=Aedes albopictus densovirus (isolate Boublik/1994)
+    #                  C=AalDNV
+    for line in combine_lines:
+        if not line:
+            continue
+        # if line is the first line of each record such as:
+        # AADNV V  648330: N=Aedes albopictus densovirus (isolate Boublik/1994)
+        if line[0] != ' ':
+            # save last complete information in map
+            if code:
+                mcf, name_seen = get_mcf_piece(code, info, seen_name_to_dcid,
+                                               CODE_TO_KINGDOM)
+                if name_seen:
+                    mcf_seen_list.append(mcf)
+                else:
+                    mcf_list.append(mcf)
+            parts = line.split('=')
+            # name_code examples: 'N', 'C', 'S'
+            name_code = parts[0][-1]
+            # name_type examples: 'Official', 'Common', 'Alternate'
+            name_type = name_type_map[name_code]
+            info[name_type] = parts[1]
+            # part_split = ['AADNV', 'V', '648330:', 'N']
+            part_split = parts[0].split()
+            code = part_split[0]
+            kingdom = part_split[1]
+            info['kingdom'] = kingdom
+            ncbiTaxonID = part_split[2][:-1]
+            info['taxonID'] = ncbiTaxonID
+        # if line is the second line of the record, such as
+        #                  C=AalDNV
+        else:
+            name_code, name = line.lstrip().split('=')
+            name_type = name_type_map[name_code]
+            info[name_type] = name
+    return mcf_seen_list, mcf_list
```

Comment:
to confirm, where is the very last "code" handled?

Reply:
Good catch that the last code wasn't imported actually! In fact last line of combine_lines is the '9ZZZZ X       1: N=root' which is not a species actually. Thus we skipped this node. Added comments.